route-snapper/src/lib.rs

```diff
@@ -1039,7 +1045,7 @@ impl Router {
             node1,
             |i| i == node2,
             |(_, _, dir_edge)| {
-                let penalty = if avoid.contains(&dir_edge.0) {
+                let penalty = if avoid.contains(&dir_edge.0) || Some(self.map.edge(dir_edge.0).name.clone()) != prev_edge_name {
```

Comment:
Same here -- even when the option is disabled, this'll clone every single edge name for the comparison. Maybe useful to split out a separate check for `match_last_road` and do the check inside there